yt/geometry/coordinates/cartesian_coordinates.py

```diff
@@ -328,34 +329,75 @@ def _ortho_pixelize(self, data_source, field, bounds, size, antialias,
+
+                    for chunk in data_source.chunks([], 'io'):
                         pixelize_sph_kernel_slice(
-                            buff_den,
+                            buff,
                             chunk[ptype, px_name],
                             chunk[ptype, py_name],
                             chunk[ptype, 'smoothing_length'],
                             chunk[ptype, 'particle_mass'],
                             chunk[ptype, 'density'],
-                            np.ones(chunk[ptype, 'density'].shape[0]),
+                            chunk[field].in_units(ounits),
                             bounds)
+                        if normalize:
+                            pixelize_sph_kernel_slice(
+                                buff_den,
+                                chunk[ptype, px_name],
+                                chunk[ptype, py_name],
+                                chunk[ptype, 'smoothing_length'],
+                                chunk[ptype, 'particle_mass'],
+                                chunk[ptype, 'density'],
+                                np.ones(chunk[ptype, 'density'].shape[0]),
+                                bounds)
 
-                if normalize:
-                    normalization_2d_utility(buff, buff_den)
+                    if normalize:
+                        normalization_2d_utility(buff, buff_den)
+
+                if smoothing_style == "gather":
+                    # for a slice we create a grid of num_pixelsxnum_pixelsx1,
+                    # for a z slice, and we swap the one dependent on the slice
+                    # axis
+                    x, y, z = self.x_axis[dim], self.y_axis[dim], dim
+
+                    buff_size = np.zeros(3, dtype="int64")
+                    buff_size[x] = size[0]
+                    buff_size[y] = size[1]
+                    buff_size[z] = 1
+
+                    buff_bounds = np.zeros(6, dtype="float64")
+                    buff_bounds[2*x:2*x+2] = bounds[0:2]
+                    buff_bounds[2*y:2*y+2] = bounds[2:4]
+                    buff_bounds[2*z] = data_source.coord
+                    buff_bounds[2*z+1] = data_source.coord
+
+                    buff_temp = np.zeros(buff_size, dtype="float64")
+
+                    pixelize_sph_gather(buff_temp, buff_bounds, self.ds,
+                                        field, ptype, normalize=normalize)
+
+                    # we swap the axes back so the axis which was sliced over
+                    # is the last axis. Then we just transpose if our x and y
+                    # were also different
+                    if z != 2:
+                        buff_temp = buff_temp.swapaxes(2, z)
+                        if x == 2:
+                            x = z
+                        else:
+                            y = z
+
+                    buff = buff_temp[:,:,0]
+                    if y < x:
+                        buff = buff.transpose()
```

Comment:
some of this may not be clear, please ping me if that is the case. It essentially just allows the slice to work in any dimension.

Initially we set up our buffer so that the the buff is of the correct shape, i.e if the x axis is sliced over that (1, n_pixels, n_pixels), then at the end we put the move these round to agree with the plot shape of (n_pixels, n_pixels, 1)

This seems a bit convoluted to me - I will look at this a bit more